From: Mathias Gibbens <gibmat@debian.org>
Date: Fri, 21 Jul 2023 16:33:51 +0800
Subject: Cherry-pick upstream fix for uint64 alignment on 32-bit
 architectures

Origin: https://github.com/go-kit/kit/pull/1007
---
 metrics/generic/generic.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/metrics/generic/generic.go b/metrics/generic/generic.go
index ebde9c8..b693345 100644
--- a/metrics/generic/generic.go
+++ b/metrics/generic/generic.go
@@ -18,9 +18,9 @@ import (
 
 // Counter is an in-memory implementation of a Counter.
 type Counter struct {
+	bits uint64 // bits has to be the first word in order to be 64-aligned on 32-bit
 	Name string
 	lvs  lv.LabelValues
-	bits uint64
 }
 
 // NewCounter returns a new, usable Counter.
@@ -81,9 +81,9 @@ func (c *Counter) LabelValues() []string {
 
 // Gauge is an in-memory implementation of a Gauge.
 type Gauge struct {
+	bits uint64 // bits has to be the first word in order to be 64-aligned on 32-bit
 	Name string
 	lvs  lv.LabelValues
-	bits uint64
 }
 
 // NewGauge returns a new, usable Gauge.
